### PR TITLE
Drop support for PHP < 7.3 and PHPUnit < 8

### PR DESCRIPTION
**Pull request by davedevelopment**
The list is getting long and unwieldy.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -31,12 +31,12 @@
...
     ],
     "require": {
-        "php": ">=5.6.0",
+        "php": ">=7.3.0",
         "lib-pcre": ">=7.0",
         "hamcrest/hamcrest-php": "~2.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "~5.7.10|~6.5|~7.0|~8.0|~9.0"
+        "phpunit/phpunit": "~8.0|~9.0"
```
Comment (localheinz) [inline composer.json:39]:
```suggestion
        "phpunit/phpunit": "^8.0.0 || ^9.0.0"
```

**Comment by davedevelopment** [comment]
@GrahamCampbell [noted](https://twitter.com/GrahamJCampbell/status/1260879743332814848) that this technically requires a major release, as we don't `require` PHPUnit.

I don't have the time or inclination to go and do all the `Towards 2.0` type things now, so I might release 2.0 as 1.0 without some of the garbage.

File: composer.json [hunk 2]
```diff
@@ -31,12 +31,12 @@
         }
     ],
     "require": {
-        "php": ">=5.6.0",
+        "php": ">=7.3.0",
```
Comment (localheinz) [inline composer.json:34]:
```suggestion
        "php": "^7.3.0",
```

**Comment by davedevelopment** [comment]
Thank you @localheinz

File: library/Mockery.php [hunk 3]
```diff
@@ -88,12 +88,9 @@ public static function builtInTypes()
             'string',
             'iterable',
             'void',
+            'object',
```
Comment (localheinz) [inline library/Mockery.php:91]: What do you think about keeping this list sorted by name?

We can probably also remove the comment `// Up to php 7` that is slightly out of view.

File: library/Mockery/Adapter/Phpunit/TestListener.php [hunk 4]
```diff
@@ -20,16 +20,36 @@
...
+    private $trait;
+
+    public function __construct()
+    {
+        $this->trait = new TestListenerTrait();
+    }
+
+
+    /**
+     * {@inheritdoc}
```
Comment (localheinz) [inline library/Mockery/Adapter/Phpunit/TestListener.php:41]: Do we need these DocBlocks?

I believe they are inherited automatically.

File: library/Mockery/Adapter/Phpunit/TestListener.php [hunk 5]
```diff
@@ -20,16 +20,36 @@
...
+    use TestListenerDefaultImplementation;
+
+    private $trait;
+
+    public function __construct()
+    {
+        $this->trait = new TestListenerTrait();
+    }
+
+
```
Comment (localheinz) [inline library/Mockery/Adapter/Phpunit/TestListener.php:39]: Looks like we have an extra empty line here, let me open up a pull request enabling a corresponding fixer.

File: library/Mockery/Adapter/Phpunit/TestListener.php (hunk 4 above)
Comment (davedevelopment) [reply library/Mockery/Adapter/Phpunit/TestListener.php]: Done

File: library/Mockery/Adapter/Phpunit/TestListener.php (hunk 5 above)
Comment (mfn) [reply library/Mockery/Adapter/Phpunit/TestListener.php]: Yeah, `no_extra_consecutive_blank_lines` isn't part of PSR2 :-/

**Pull request merged**